cura/CuraPackageManager.py

```diff
@@ -0,0 +1,357 @@
+                            "display_name": plugin_metadata["plugin"]["name"],
+                            "description": plugin_metadata["plugin"].get("description"),
+                            "package_version": plugin_metadata["plugin"]["version"],
+                            "cura_version": int(plugin_metadata["plugin"]["api"]),
+                            "website": "",
+                            "author": {
+                                "name": plugin_metadata["plugin"].get("author", ""),
+                                "email": "",
+                                "website": "",
+                            },
+                            "tags": ["plugin"],
+                            }
+        return package_metadata
+
+    # Checks if the given package is installed.
+    def isPackageInstalled(self, package_id: str) -> bool:
+        return self.getInstalledPackageInfo(package_id) is not None
+
+    # Schedules the given package file to be installed upon the next start.
+    @pyqtSlot(str)
+    def installPackage(self, filename: str) -> None:
+        # Get package information
+        package_info = self.getPackageInfo(filename)
+        package_id = package_info["package_id"]
+
+        has_changes = False
+        # Check the delayed installation and removal lists first
+        if package_id in self._to_remove_package_set:
+            self._to_remove_package_set.remove(package_id)
+            has_changes = True
+
+        # Check if it is installed
+        installed_package_info = self.getInstalledPackageInfo(package_info["package_id"])
+        to_install_package = installed_package_info is None  # Install if the package has not been installed
+        if installed_package_info is not None:
+            # Compare versions and only schedule the installation if the given package is newer
+            new_version = package_info["package_version"]
+            installed_version = installed_package_info["package_version"]
+            if VersionTools.compareSemanticVersions(new_version, installed_version) > 0:
+                Logger.log("i", "Package [%s] version [%s] is newer than the installed version [%s], update it.",
+                           package_id, new_version, installed_version)
+                to_install_package = True
+
+        if to_install_package:
+            Logger.log("i", "Package [%s] version [%s] is scheduled to be installed.",
+                       package_id, package_info["package_version"])
+            # Copy the file to cache dir so we don't need to rely on the original file to be present
+            package_cache_dir = os.path.join(os.path.abspath(Resources.getCacheStoragePath()), "cura_packages")
+            if not os.path.exists(package_cache_dir):
+                os.makedirs(package_cache_dir, exist_ok=True)
+
+            target_file_path = os.path.join(package_cache_dir, package_id + ".curapackage")
+            shutil.copy2(filename, target_file_path)
```

Comment:
This is not atomic, so you need to acquire the configuration file lock before doing this, or you'll have corrupt files if two instances of Cura are writing at the same time.